### ISSUE TITLE
Incorporate DCaPST

### ISSUE THREAD

**Issue by jbrider** [APSIMInitiative/APSIMClassic#1770]
Incorporate changes to enable DCaPST 
(Diurnal Canopy Photosynthesis-Stomatal Conductance Simulator)

Wu, A., Doherty, A., Wang, E., and Hammer, G.L. (2018) Simulating daily field crop canopy photosynthesis: an integrated software package. Functional Plant Biology, Functional Plant Biology, 45: 362-377.  https://doi.org/10.1071/FP17225

**Comment by wangyakai01** [comment]
Hello, I wonder how many crops is the DCaPST module suitable for? I know that it has been incorporated with the wheat, maize, and sorghum crop modules. Can it be coupled with other crop modules? I have tried to couple the DCaPST module with the soybean using the version of APSIM7.8 and found that it works, but when I used the latest version of APSIM710-r4207 to couple the DCaPS-C3 module with the soybean, such errors sometimes occur:
```
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
                      APSIM  Fatal  Error
                      -------------------
     Gbh cannot be 0 paddock.DCaPS-C3.dodcapst
     Component name: DCaPS-C3
     
     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
```

I checked the script code of "DCaPS-C3" module and found that if the code  "MyPaddock.Set("PsModelName", "dcapst");" is commented out, such as
```
 if (!dcapstON && lai >= LAITrigger)
    {
         dcapstON = true;
        // MyPaddock.Set("PsModelName", "dcapst");
     }

```
 no error will be reported after running. However, the output variable of BIOshootDAY will always be 0. It is said that the DCaPST module is not working. What is the reason for this? @jbrider
